app/models/cart.rb

```diff
@@ -0,0 +1,18 @@
+class Cart < ApplicationRecord
+  has_many :order_items, :dependent => :destroy
+
+  def add_product(product_id)
+    current_item = order_items.find_by_product_id(product_id) 
+    if current_item
+      current_item.quantity += 1
+    else
+      current_item = order_items.build(product_id: product_id)
```

Comment:
So, long story short, in the ```create``` function from ```order_items_controller.rb```, this function is called in order to do a query on the database to find if the product's id which we want to add to the cart is present in any ```order_item``` where the ```cart_id``` is equal to the user's ```cart_id```. After getting that information it either updates the quantity, or creates another one with a default quantity of 1. This function only returns the modified/newly created item, which is later saved in ```order_items_controller.rb``` @line 6 where ```@order_item.save``` is called and it's added/modified @database. Building it this way assures that ```order_item.cart_id``` is set to the ```current user's cart.id``` ✓
Bump me up if i missed anything or more details are needed ❤️.

Reply:
Hmm, so with this approach, what happens if I add a product to the cart and login from a different device? Will it display my cat items  there?

Reply:
In the current state, technically no. This is because upon loading the page, the ```set_cart``` action either loads the cart with ```cart.id = session[:cart_id]``` from a previous browsing session or creates a new one. Using the cart in the current state does not require the user to be logged in and information is stored in session rather than on user. 
My question here is the following: should the user be required to log in before adding items to cart & use the cart, or, if not, how should the merging occur between current cart items(before login) and saved items loaded upon login from the previous session?